deals/deals.go

```diff
@@ -225,6 +234,195 @@ func (m *Module) Watch(ctx context.Context, proposals []cid.Cid) (<-chan DealInf
 	return ch, nil
 }
 
+// FinalDealRecords returns a list of all finalized storage deals.
+func (m *Module) FinalDealRecords() ([]DealRecord, error) {
+	return m.store.getFinalDeals()
+}
+
+// PendingDealRecords returns a list of all pending storage deals.
+func (m *Module) PendingDealRecords() ([]DealRecord, error) {
+	return m.store.getPendingDeals()
+}
+
+// AllDealRecords returns a list of all finalized and pending deals.
+func (m *Module) AllDealRecords() ([]DealRecord, error) {
+	ret, err := m.store.getFinalDeals()
+	if err != nil {
+		return nil, err
+	}
+	pending, err := m.store.getPendingDeals()
+	if err != nil {
+		return nil, err
+	}
+	ret = append(ret, pending...)
+	sort.Slice(ret, func(i, j int) bool {
+		if ret[i].Pending || ret[j].Pending {
+			return ret[i].Time < ret[j].Time
+		}
+		if ret[i].DealInfo.ActivationEpoch < ret[j].DealInfo.ActivationEpoch {
+			return true
+		}
+		if ret[i].DealInfo.ActivationEpoch > ret[j].DealInfo.ActivationEpoch {
+			return false
+		}
+		return ret[i].Time < ret[j].Time
+	})
+	return ret, nil
+}
+
+// RetrievalRecords returns a list of all retrievals.
+func (m *Module) RetrievalRecords() ([]RetrievalRecord, error) {
+	return m.store.getRetrievals()
+}
+
+// Close closes the deals Module.
+func (m *Module) Close() error {
+	return m.store.close()
+}
+
+func (m *Module) initPendingDeals() {
```

Comment:
This gets calls on deal module creation. Goal here is to return as soon as possible by spawning off all sub routines in go routines. We query for any previously pending deals, calculate the time remaining to continue monitoring each one, then begin monitoring each one. Or, if the pending deal's monitoring period has timed out, we finalize it immediately.